kafkabp/config.go

```diff
@@ -138,6 +139,10 @@ func (cfg *ConsumerConfig) NewSaramaConfig() (*sarama.Config, error) {
 	// Return any errors that occurred to the Errors channel.
 	c.Consumer.Return.Errors = true
 
+	// Set the commit frequency to 5s
+	c.Consumer.Offsets.AutoCommit.Enable = true
+	c.Consumer.Offsets.AutoCommit.Interval = 5 * time.Second
```

Comment:
:no_bell: one way to make it more flexible is to define it as a global, exported variable (e.g.`var CommitInterval = 5 * time.Second`), so if 5s doesn't work for some services they an override it by changing the value of the global variable before creating their consumers. but we don't necessarily need to do that now and can wait until such need actually arise.

Reply:
Agree it might be nice to have this configurable, especially since it appears most code calls `NewConsumer` rather than this directly so (from a cursory glance) I'm not sure if it's possible to modify as-is.

https://github.com/reddit/baseplate.go/blob/088a0beef0815a5961b5b6e0efccc025ae318c0d/kafkabp/consumer.go#L97-L101

Reply:
I added it to the ConsumerConfig, let me know what you think - yall are right that some consumers might want to change this value (though as Matt says below, imo that's indicative of other problems with the consumer). let me know what you think.

Reply:
depending on how often/rarely services actually want to change it, we don't necessarily want it to be part of config.

part of the config is suitable for cases that this is more likely to be changed by services (say, ~40% of the services would want to use a non-default value), while defining a global variable is more like "in most cases you don't want to change it but if you really do and know what you are doing you can change it" case.